tests/lib/rules/no-potential-component-option-typo.js

```diff
@@ -0,0 +1,245 @@
+      <script>
+      export default {
+      }
+      </script>
+      `.trim(),
+      options: [{ presets: ['vue'] }]
+    },
+    // test if give preset and the potentialTypoList length is zero, just for 100% test cover
+    {
+
+      filename: 'test.vue',
+      code: `
+      <script>
+      export default {
+        data() {}
+      }
+      </script>
+      `.trim(),
+      options: [{ presets: ['vue'] }]
+    }
+    // give me some code that won't trigger a warning
+  ],
+  invalid: [
+    {
+      filename: 'test.vue',
+      code: `
+      <script>
+      export default {
+        dat: {},
+        method: {}
+      }
+      </script>
+      `.trim(),
+      errors: [
+        {
+          message: "'dat' may be a typo, which is similar to option [data].",
+          suggestions: [
+            {
+              desc: `Replace property 'dat' to 'data'`,
+              output: `
+      <script>
+      export default {
+        data: {},
+        method: {}
+      }
+      </script>
+              `.trim()
+            }
+          ]
+        },
+        {
+          message: `'method' may be a typo, which is similar to option [methods].`,
+          suggestions: [
+            {
+              desc: `Replace property 'method' to 'methods'`,
+              output: `
+      <script>
+      export default {
+        dat: {},
+        methods: {}
+      }
+      </script>
+              `.trim()
+            }
+          ]
+        }
+      ],
+      options: [{ custom: ['data', 'methods'] }]
+    },
+    // test if user define custom rule is duplicate with presets
+    {
+      filename: 'test.vue',
+      code: `
+      <script>
+      export default {
+        dat: {},
+        method: {}
+      }
+      </script>
+      `.trim(),
+      errors: [
+        {
+          message: "'dat' may be a typo, which is similar to option [data].",
+          suggestions: [
+            {
+              desc: `Replace property 'dat' to 'data'`,
+              output: `
+      <script>
+      export default {
+        data: {},
+        method: {}
+      }
+      </script>
+              `.trim()
+            }
+          ]
+        },
+        {
+          message: `'method' may be a typo, which is similar to option [methods].`,
+          suggestions: [
+            {
+              desc: `Replace property 'method' to 'methods'`,
+              output: `
+      <script>
+      export default {
+        dat: {},
+        methods: {}
+      }
+      </script>
+              `.trim()
+            }
+          ]
+        }
+      ],
+      options: [{ custom: ['data', 'methods'], presets: ['all'] }]
+    },
+    // test if report correctly, only have preset option
+    {
+
+      filename: 'test.vue',
+      code: `
+      <script>
+      export default {
+        dat: {},
+        method: {}
+      }
+      </script>
+      `.trim(),
+      errors: [
+        {
+          message: "'dat' may be a typo, which is similar to option [data].",
```

Comment:
Could you test the location of the error using the `line` and `column` properties?